ntoskrnl/io/pnpmgr/pnpres.c

```diff
@@ -1184,19 +1218,24 @@ IopCheckForResourceConflict(
 {
    ULONG i, ii;
    BOOLEAN Result = FALSE;
+   PCM_FULL_RESOURCE_DESCRIPTOR FullDescriptor;
+   PCM_PARTIAL_RESOURCE_DESCRIPTOR ResDesc;
 
+   FullDescriptor = &ResourceList1->List[0];
    for (i = 0; i < ResourceList1->Count; i++)
    {
-      PCM_PARTIAL_RESOURCE_LIST ResList = &ResourceList1->List[i].PartialResourceList;
+      PCM_PARTIAL_RESOURCE_LIST ResList = &FullDescriptor->PartialResourceList;
+      FullDescriptor = CmiGetNextResourceDescriptor(FullDescriptor);
+
+      ResDesc = &ResList->PartialDescriptors[0];
       for (ii = 0; ii < ResList->Count; ii++)
       {
-         PCM_PARTIAL_RESOURCE_DESCRIPTOR ResDesc = &ResList->PartialDescriptors[ii];
-
          Result = IopCheckResourceDescriptor(ResDesc,
                                              ResourceList2,
                                              Silent,
                                              ConflictingDescriptor);
          if (Result) goto ByeBye;
+         ResDesc = CmiGetNextPartialDescriptor(ResDesc);
```

Comment:
Why not put this after `ii++` into for statement? Just to be consistent with other code